src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -11,6 +16,23 @@ package dsls.regex
 abstract class RegularExpression {
   /** returns true if the given string matches this regular expression */
   def matches(string: String) = RegexMatcher.matches(string, this)
+  /** returns a union of two regular expression */
+  def ||(other: RegularExpression) = Union(this, other)
+  /** returns the concatenation of two regular expressions */
+  def ~(other: RegularExpression) = Concat(this, other)
+  /** returns the stared version of the regular expression */
+  def <*>() = Star(this)
+  /** returns a regex which requires one copy of the regex
+      followed by any number of copies following (via Star) */
+  def <+>() = Concat(this, Star(this))
+  /** returns a regex which is n concatenations of the regex */
+  def apply(n: Int): RegularExpression = {
+    if (n == 1) {
```

Comment:
Honestly, it seems like `{0}` should be an error - I can't think of a case where this would be anything except for a typo.